src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -0,0 +1,323 @@
+            shaToFlow,
+            discardPatches,
+            cancellationToken);
+
+        if (branchName is null)
+        {
+            // TODO: We should still probably update package versions or at least try?
+            // Should we clean up the repos?
+            return null;
+        }
+
+        await UpdateVersionDetailsXml(targetRepo, shaToFlow, cancellationToken);
+        return branchName;
+    }
+
+    protected override async Task<string?> SameDirectionFlowAsync(
+        SourceMapping mapping,
+        string shaToFlow,
+        NativePath repoPath,
+        Codeflow lastFlow,
+        bool discardPatches,
+        CancellationToken cancellationToken)
+    {
+        var isBackflow = lastFlow is Backflow;
+        var shortShas = $"{Commit.GetShortSha(lastFlow.SourceSha)}-{Commit.GetShortSha(shaToFlow)}";
+        var branchName = $"codeflow/{lastFlow.GetType().Name.ToLower()}/{shortShas}";
+        var targetRepo = isBackflow ? repoPath : _vmrInfo.VmrPath;
+        var submoduleExclusions = _sourceManifest.Submodules
+            .Where(s => s.Path.StartsWith(mapping.Name + '/'))
+            .Select(s => s.Path.Substring(mapping.Name.Length + 1))
+            .Select(VmrPatchHandler.GetExclusionRule)
+            .ToList();
+
+        // When flowing from the VMR, ignore all submodules
+        List<VmrIngestionPatch> patches = await _vmrPatchHandler.CreatePatches(
+            _vmrInfo.TmpPath / (mapping.Name + "-backflow-" + shortShas + ".patch"),
+            lastFlow.VmrSha,
+            shaToFlow,
+            path: null,
+            filters: submoduleExclusions,
+            relativePaths: true,
+            workingDir: _vmrInfo.GetRepoSourcesPath(mapping),
+            applicationPath: null,
+            cancellationToken);
+
+        if (patches.Count == 0 || patches.All(p => _fileSystem.GetFileInfo(p.Path).Length == 0))
+        {
+            _logger.LogInformation("There are no new changes for {mappingName} between {sha1} and {sha2}",
+                isBackflow ? "VMR" : mapping.Name,
+                lastFlow.SourceSha,
+                shaToFlow);
+
+            if (discardPatches)
+            {
+                foreach (VmrIngestionPatch patch in patches)
+                {
+                    _fileSystem.DeleteFile(patch.Path);
+                }
+            }
+
+            return null;
+        }
+
+        _logger.LogInformation("Created {count} patch(es)", patches.Count);
+
+        await _localGitClient.CheckoutAsync(targetRepo, lastFlow.TargetSha);
+        await _workBranchFactory.CreateWorkBranchAsync(targetRepo, branchName);
+
+
+        // TODO: Remove VMR patches before we create the patches
+
+        try
+        {
+            foreach (VmrIngestionPatch patch in patches)
+            {
+                await _vmrPatchHandler.ApplyPatch(patch, targetRepo, cancellationToken);
+
+                if (discardPatches)
+                {
+                    _fileSystem.DeleteFile(patch.Path);
```

Comment:
Consider: pass the discard to the ApplyPatch() method and do the cleanup there. It will save the if and would logically fit there.